### ISSUE TITLE
java.lang.OutOfMemoryError: Java heap space when using GeMoma 

### ISSUE THREAD

**Issue by xiongqian123456789** [Jstacs/Jstacs#49]
Hello! 
I used GeMoma by with given external annotation external.gff, the command is as follows:

java -jar /newlustre/home/xiongqian/software/annotation/gemoma/GeMoMa-1.9.jar CLI GeMoMaPipeline threads=24 tblastn=false GeMoMa.Score=ReAlign AnnotationFinalizer.r=NO o=true t=D2005012427.Euk.IDs.fa outdir=test s=own i=1 a=GCA_000143455.1.gff g=GCA_000143455.1.fas  s=own i=2 a=GCA_001584585.1.gff g=GCA_001584585.1.fas  s=own i=3 a=GCA_030144725.1.gff g=GCA_030144725.1.fas s=own i=4 a=GCA_030272155.1.gff g=GCA_030272155.1.fas ID=august e=D2005012427.augustus.gff3

And get the error is：
Problem while gene: gene-QJQ45_015695_
GeMoMa for species 2 (3) split=20 throws an Exception
java.lang.Exception: Forwarding java.lang.OutOfMemoryError: Java heap space
        at de.jstacs.algorithms.alignment.Alignment.computeAlignment(Alignment.java:248)
        at projects.gemoma.GeMoMa$MyAlignment.computeAlignment(GeMoMa.java:4707)
        at de.jstacs.algorithms.alignment.Alignment.getAlignment(Alignment.java:185)
        at de.jstacs.algorithms.alignment.Alignment.getAlignment(Alignment.java:167)
        at projects.gemoma.GeMoMa$TranscriptPredictor.computeTranscript(GeMoMa.java:2636)
        at projects.gemoma.GeMoMa$TranscriptPredictor.compute(GeMoMa.java:2336)
        at projects.gemoma.GeMoMa.run(GeMoMa.java:1012)
        at projects.gemoma.GeMoMaPipeline$JGeMoMa.doJob(GeMoMaPipeline.java:1988)
        at projects.gemoma.GeMoMaPipeline$FlaggedRunnable.run(GeMoMaPipeline.java:1375)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

GeMoMa for species 2 (3) split=16 throws an Exception
GeMoMa for species 2 (3) split=7 throws an Exception
GeMoMa for species 2 (3) split=10 throws an Exception
GeMoMa for species 2 (3) split=14 throws an Exception

Really apprieciated for your help!

**Comment by JensKeilwagen** [comment]
Hi,

Thanks a lot for your interest in GeMoMa.

As the Exception states its a memory issue. You should try to increase allowed memory when starting java (-Xmx).
We also have this issue in the FAQs (http://www.jstacs.de/index.php/GeMoMa#Frequently_asked_questions).

best regards, Jens

**Comment by xiongqian123456789** [comment]
Dear Jens,
Really Appreciated for your reply!
 
I used the new command to run Gemoma, and met the new error as followed:
java -Xms20G -Xmx200G -jar /newlustre/home/xiongqian/software/annotation/gemoma/GeMoMa-1.9.jar CLI GeMoMaPipeline threads=40 tblastn=false AnnotationFinalizer.r=NO o=true t=D2203116285.Euk.IDs.fa outdir=265 p=true pc=true pgr=true s=own i=1 a=GCA_000143455.1.gff g=GCA_000143455.1.fas s=own i=2 a=GCA_001584585.1.gff g=GCA_001584585.1.fas s=own i=3 a=GCA_030144725.1.gff g=GCA_030144725.1.fas s=own i=4 a=GCA_030272155.1.gff g=GCA_030272155.1.fas ID=august e=/newlustre/home/xiongqian/annotation/gemoma/refer-gemoma/gff_augustus_c/D2203116285.augustus.gff3


There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (mmap) failed to map 13420199936 bytes for committing reserved memory.
# Possible reasons:
#   The system is out of physical RAM or swap space
#   In 32 bit mode, the process size limit was hit
# Possible solutions:
#   Reduce memory load on the system
#   Increase physical memory or swap space
#   Check if swap backing store is full
#   Use 64 bit Java on a 64 bit OS
#   Decrease Java heap size (-Xmx/-Xms)
#   Decrease number of Java threads
#   Decrease Java thread stack sizes (-Xss)
#   Set larger code cache with -XX:ReservedCodeCacheSize=
# This output file may be truncated or incomplete.
#
#  Out of Memory Error (os_linux.cpp:2627), pid=735, tid=47319578621696
#
# JRE version: OpenJDK Runtime Environment (8.0_65-b17) (build 1.8.0_65-b17)
# Java VM: OpenJDK 64-Bit Server VM (25.65-b01 mixed mode linux-amd64 )
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

**Comment by JensKeilwagen** [comment]
Hi,

it seems that you're running GeMoMa on a 32-bit system. 32-bit systems have a natural limit of RAM which is 4 Gb (https://www.maplesystems.com/supportcenter/faq?qid=393&q=What%20are%20the%20RAM%20limits%20of%20a%2032-bit%20operating%20system%3F). This is no GeMoMa bug/issue.

I would recommend to run GeMoMa in a 64-bit environment.

best regards, Jens

**Comment by xiongqian123456789** [comment]
Thank you! 
But the Administrator told me the environment anf JDK is 64-bit. 

(gemoma) [xiongqian@mu01 chlo-chlorophyceae]$ uname -m
x86_64


(gemoma) [xiongqian@mu01 chlo-chlorophyceae]$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Comment by xiongqian123456789** [comment]
Hello.

I tried several times, showed the same error, and the gemoma was installed by hand and bioconda.

**Comment by xiongqian123456789** [comment]
Hello.

I tried several times, showed the same error, and the gemoma was installed by hand and bioconda.

**Comment by JensKeilwagen** [comment]
Hi,

Hi I checked again your posts, it could be that you are working on a 64 bit system (`uname -m` (=x86_64) and `java -version` (=OpenJDK 64-Bit Server)) but maybe there is some other issue. In your report, I found _Decrease Java heap size (-Xmx/-Xms)_. Hence, it could be that you just requested too much RAM. Based on your commandline you asked for **-Xms20G -Xmx200G**. Hence, we need to check whether your system provides so much RAM.
 
Just to be sure, please do the following on the command line of your **compute node** and report it to me:

1. Determine whether you have a 32 or 64 bit system: `uname -m`
2. Determine the Java version: `java -version`
3. Determine the total RAM on your system: `grep MemTotal /proc/meminfo`

best regards, Jens

**Comment by xiongqian123456789** [comment]
Hi Jens，
Thank you very much !

The three nodes were:
(base) [xiongqian@mu01 Orthogroups]$ uname -m
x86_64
(base) [xiongqian@mu01 Orthogroups]$ java -version
openjdk version "1.8.0_65"
OpenJDK Runtime Environment (build 1.8.0_65-b17)
OpenJDK 64-Bit Server VM (build 25.65-b01, mixed mode)
(base) [xiongqian@mu01 Orthogroups]$ grep MemTotal /proc/meminfo
MemTotal:       32623108 kB

And I use pbs to submit the task in cluster, the node i used was: 


(base) [xiongqian@mu01 Orthogroups]$ pbsnodes
fat01
     state = free
     power_state = Running
     np = 40
     properties = fatPAR
     ntype = cluster
     status = rectime=1703230509,macaddr=6c:92:bf:14:d8:7a,cpuclock=Fixed,varattr=,jobs=,state=free,size=783772724kb:830956888kb,netload=187438545716,gres=,loadave=0.00,ncpus=40,physmem=1056526600kb,availmem=1044261892kb,totmem=1089294596kb,idletime=487077,nusers=3,nsessions=5,sessions=4616 30722 133635 133639 133707,uname=Linux fat01 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003

fat02
     state = down
     power_state = Running
     np = 40
     properties = fatPAR
     ntype = cluster
     status = rectime=1696734100,macaddr=6c:92:bf:14:d7:6a,cpuclock=Fixed,varattr=,jobs=,state=free,size=781324248kb:830956888kb,netload=11740809,gres=,loadave=1.14,ncpus=40,physmem=1056526600kb,availmem=1081366696kb,totmem=1089294596kb,idletime=99,nusers=2,nsessions=4,sessions=5631 6097 7065 7069,uname=Linux fat02 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003

fat03
     state = job-exclusive
     power_state = Running
     np = 40
     properties = fatPAR
     ntype = cluster
     jobs = 0-39/131529.mu01
     status = rectime=1703230522,macaddr=6c:92:bf:14:d5:22,cpuclock=Fixed,varattr=,jobs=131529.mu01,state=free,size=786845372kb:830956888kb,netload=145627277657,gres=,loadave=0.00,ncpus=40,physmem=1056526600kb,availmem=1021353596kb,totmem=1089294596kb,idletime=20940010,nusers=2,nsessions=4,sessions=4200 4213 123760 123764,uname=Linux fat03 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003

fat04
     state = free
     power_state = Running
     np = 48
     properties = fatPAR
     ntype = cluster
     jobs = 0-39/131529.mu01
     status = rectime=1703230530,macaddr=08:94:ef:54:ed:58,cpuclock=Fixed,varattr=,jobs=131529.mu01(cput=1706884,energy_used=0,mem=975322348kb,vmem=1326395616kb,walltime=36025,session_id=112944),state=free,size=12464260kb:52403200kb,netload=128160984766,gres=,loadave=801.61,ncpus=48,physmem=990072304kb,availmem=39348020kb,totmem=990072304kb,idletime=24198323,nusers=1,nsessions=1,sessions=112944,uname=Linux fat04 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003

fat05
     state = job-exclusive
     power_state = Running
     np = 48
     properties = newfatPAR
     ntype = cluster
     jobs = 0-47/131567.mu01
     status = rectime=1703230513,macaddr=08:94:ef:54:f0:98,cpuclock=Fixed,varattr=,jobs=131567.mu01(cput=82865,energy_used=0,mem=20105716kb,vmem=61748452kb,walltime=2182,session_id=357262),state=free,size=11914680kb:52403200kb,netload=35019895527491,gres=,loadave=44.40,ncpus=48,physmem=1056128456kb,availmem=1009854200kb,totmem=1060322756kb,idletime=18485758,nusers=2,nsessions=2,sessions=357262 388979,uname=Linux fat05 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003

gpu01
     state = free
     power_state = Running
     np = 20
     properties = gpuPAR
     ntype = cluster
     status = rectime=1703230513,macaddr=6c:92:bf:23:6e:94,cpuclock=Fixed,varattr=,jobs=,state=free,size=1841030200kb:1935274740kb,netload=2072649772668,gres=,loadave=0.00,ncpus=20,physmem=131747088kb,availmem=145817156kb,totmem=148524300kb,idletime=8225286,nusers=0,nsessions=0,uname=Linux gpu01 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003

**Comment by JensKeilwagen** [comment]
Hi,

thanks a lot. I did not completely understand as you provided so much information. It would be good to only have the information of the compute node that was used. The easiest way would be to include the three commands at the beginning of your script that starts GeMoMa.

As far as I see there are nodes with different amount of RAM:

`MemTotal: 32623108 kB` (=32GB)
`fat01 ... totmem=1089294596kb` (=1089GB)
`fat02 ... totmem=1089294596kb`
`fat03 ... totmem=1089294596kb`
`fat04 ... totmem=990072304kb`
`fat05 ... totmem=1060322756kb`
`gpu01 ... totmem=148524300k` (=148GB)

So it could be that the job is running on a node that has not enough memory, as you requested `-Xmx200G`.

best regards, Jens

**Comment by xiongqian123456789** [comment]
Thank you very much, Jens!

I will try again by a larger RAM. And i do not understant how to set a suitable value for -Xmx, could you give me some suggestion?

**Comment by JensKeilwagen** [comment]
The amount of RAM is hard to estimate it depends on the input. Especially the target genome size, but also on the annotation of the related species.
I think you either need to run it on a node with more RAM or you need to decrease the RAM you requested as it might be an overestimation.

**Comment by xiongqian123456789** [comment]
Hi Jens,

although I run the command with larger RAM as :
java -Xms20G -Xmx100G -jar /data_group/miaowei/luoshuai/software/miniconda3/envs/gemoma/share/gemoma-1.9-0/GeMoMa-1.9.jar CLI GeMoMaPipeline threads=40 tblastn=false GeMoMa.Score=ReAlign AnnotationFinalizer.r=NO o=true t=D2203116285.Euk.IDs.fa outdir=test85 s=own i=1 a=GCA_000143455.1.gff g=GCA_000143455.1.fas  s=own i=2 a=GCA_001584585.1.gff g=GCA_001584585.1.fas  s=own i=3 a=GCA_030144725.1.gff g=GCA_030144725.1.fas s=own i=4 a=GCA_030272155.1.gff g=GCA_030272155.1.fas ID=august e=D2203116285.augustus.gff3


The log file are as the attachment: 

Thank you very much!

**Comment by JensKeilwagen** [comment]
Hi,

I'm sorry, I've been on Christmas holidays and I'm just back in office.
I saw your post but did not see the attachment.

best regards, Jens

**Comment by jwli-code** [comment]
Hello, I have also encountered the same problem. I have multiple reference genomes, among which two reference genomes have also reported the same problem of insufficient memory for java operation. I wonder if the running time and memory of this software are calculated according to the size of the genome, and why some large genomes only run about 20G of memory. Some small genomes run for a long time and still report java errors. Here is my running code. #java -jar /home/jwli/miniconda3/share/gemoma-1.9-0/GeMoMa-1.9.jar CLI Extractor \
#a=$ref_gff_path \
#g=$ref_fa_path \
#outdir=./$ref_prefix
# blast cds to target genome
#mmseqs createdb $ref_prefix/cds-parts.fasta  $ref_prefix/queryDB

#mmseqs easy-search $ref_prefix/cds-parts.fasta TargetGenomeDB $ref_prefix/resultDB tmp --format-output query,target,pident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits,empty,raw,nident,empty,empty,empty,qframe,tframe,qaln,taln,qlen,tlen --threads 30

# get gff prediction from this ref genome
java -jar -Xms200G -Xmx400G /home/jwli/miniconda3/share/gemoma-1.9-0/GeMoMa-1.9.jar CLI GeMoMa s=$ref_prefix/resultDB \
t=Brassica_napus.Darmor.v4.1.genome.fa \
c=$ref_prefix/cds-parts.fasta r=1 m=100000 e=100 \
a=$ref_prefix/assignment.tabular \
tag=mRNA sort=true  \
outdir=$ref_prefix
@JensKeilwagen @xiongqian123456789

**Comment by JensKeilwagen** [comment]
Hi jwli-code,

Thanks a lot for your interest in GeMoMa. 

I'm not sure whether the problem is the same. It seems that you are running the individual steps independently, while xiongqian123456789 is running the GeMoMaPipeline. I also did not see your error message.

I would be happy to help both of you, but I need more information.
It would be good to get the complete error message and at best a small example where the error occurs that I can use to trace the problem.

best regards, Jens

**Comment by jwli-code** [comment]
hi,
I solved the above problem by further increasing the memory, but I had a question, if I have multiple reference genomes and I want to bring in the second generation transcriptome data BAM, should I run each one in BAM if I run separately? Does this consume the same computing resources as GeMoMaPipeline? And I don't know how to bring in the transcriptome data separately, what are the specific commands?
_Here are my orders to run alone_
```
# get gff prediction from this ref genome
java -jar -Xms20G -Xmx400G /home/jwli/miniconda3/share/gemoma-1.9-0/GeMoMa-1.9.jar CLI GeMoMa s=$ref_prefix/resultDB \
t=Brassica_napus.ZS11.v0.genome.fa \
c=$ref_prefix/cds-parts.fasta r=1 m=100000 e=100 \
a=$ref_prefix/assignment.tabular \
tag=mRNA sort=true  \
i=./01_ERE_intron/introns.gff \
outdir=$ref_prefix
```
best regards, jwli

**Comment by JensKeilwagen** [comment]
Hi jwli,
I'm not sure whether your question should be handled as a separate issue. It does not fit the current issue title anymore.

I also have to admit that I do not complete understand all of your questions. I'll try to answer point by point:
- You can decide which read mapper you like to use. For RNAseq we have good experience with STAR but it is up to you.
- The transcriptome data needs to be mapped against the target genome and not the references. Hence, it is independent of the reference species.
- If you have transcriptome data you can use introns **and coverage** from these data to improve the predictions. It also to predict UTRs.
- If you have multple independet transcriptome data sets you can include all of them (https://github.com/Jstacs/Jstacs/issues/4)
- If you run GeMoMaPipeline, you can call it once with all of your reference species. Internally, GeMoMaPipeline runs independent runs of GeMoMa for each reference and combines the predictions using GAF. Thereby GeMoMaPipeline allows to run things in parallel and avoids to do some tasks multiple times. The memory consumption scales with the number of threads.

best regards, Jens

**Comment by xiongqian123456789** [comment]
Hi Jens,

although I run the command with larger RAM as :
java -Xms20G -Xmx100G -jar /data_group/miaowei/luoshuai/software/miniconda3/envs/gemoma/share/gemoma-1.9-0/GeMoMa-1.9.jar CLI GeMoMaPipeline threads=40 tblastn=false GeMoMa.Score=ReAlign AnnotationFinalizer.r=NO o=true t=D2203116285.Euk.IDs.fa outdir=test85 s=own i=1 a=GCA_000143455.1.gff g=GCA_000143455.1.fas s=own i=2 a=GCA_001584585.1.gff g=GCA_001584585.1.fas s=own i=3 a=GCA_030144725.1.gff g=GCA_030144725.1.fas s=own i=4 a=GCA_030272155.1.gff g=GCA_030272155.1.fas ID=august e=D2203116285.augustus.gff3

The log file are as the attachment: 

Really appreciate for your help!













At 2024-01-08 19:54:18, "keili" ***@***.***> wrote:

Hi jwli,
I'm not sure whether your question should be handled as a separate issue. It does not fit the current issue title anymore.

I also have to admit that I do not complete understand all of your questions. I'll try to answer point by point:

You can decide which read mapper you like to use. For RNAseq we have good experience with STAR but it is up to you.
The transcriptome data needs to be mapped against the target genome and not the references. Hence, it is independent of the reference species.
If you have transcriptome data you can use introns and coverage from these data to improve the predictions. It also to predict UTRs.
If you have multple independet transcriptome data sets you can include all of them (#4)
If you run GeMoMaPipeline, you can call it once with all of your reference species. Internally, GeMoMaPipeline runs independent runs of GeMoMa for each reference and combines the predictions using GAF. Thereby GeMoMaPipeline allows to run things in parallel and avoids to do some tasks multiple times. The memory consumption scales with the number of threads.

best regards, Jens

**Comment by JensKeilwagen** [comment]
Hi,

I'm sorry to hear that you still have problems. However, I don't see any attachment. I also don't have the data for a minimal example that shows this error. So currently, I cannot recommend anything. Please provide the log-file and data for a minimal example. 

I would be happy to help you solve the problem.

best regards, Jens

**Comment by JensKeilwagen** [comment]
Thanks for sending the files via email. It seems that it is not a memory issue.
The error log states:
```
Exception in thread "main" java.lang.RuntimeException: Did not finish as intended. java.lang.NullPointerException: null
	at projects.gemoma.GeMoMaPipeline.run(GeMoMaPipeline.java:1038)
	at projects.gemoma.GeMoMaModule.run(GeMoMaModule.java:94)
	at de.jstacs.tools.ui.cli.CLI.run(CLI.java:426)
	at projects.gemoma.GeMoMa.main(GeMoMa.java:399)
```

I assume it could be that the synteny check is making the problems as Augustus predictions cannot be used for synteny check (https://github.com/Jstacs/Jstacs/issues/43, https://github.com/Jstacs/Jstacs/issues/21). Please try to run with `sc=false`

However, I also noticed that the result are somehow wired. Although you have enough transcripts extracted from the reference species (12-24K transcripts per reference species), the GAF module filters out most of them:

```
all: 605809
filtered: 16144
clustered: 9027

...

input	filtered transcripts	final transcripts supported	final genes supported
input 0	106	89	89
input 1	203	158	157
input 2	248	72	72
input 3	166	117	117
input 4 (august)	15594	8839	8839
```

This seems to be wired. Less than 250 transcripts per reference species passed the filter. It seems that you haven't modified the filter, i.e. it is the default value. With default value (`start=='M' and stop=='*' and (isNaN(score) or score/aa>=0.75)`) it filters for having a start and a stop codon and for some conservation (score/aa>=0.75). I would recommend to check why so many predictions were discarded.

I would recommend to run GeMoMaPipeline again with additional parameter setting `sc=false`. You already used `o=true`.  This allows to run GAF with different parameter settings after the GeMoMaPipeline.

**Comment by xiongqian123456789** [comment]
Hi Jens,


Thank you very much! It has been run successfully with the parameter setting sc=false. 


Really appreciate for your help! 
















At 2024-01-10 16:39:00, "keili" ***@***.***> wrote:

Thanks for sending the files via email. It seems that it is not a memory issue.
The error log states:

Exception in thread "main" java.lang.RuntimeException: Did not finish as intended. java.lang.NullPointerException: null
	at projects.gemoma.GeMoMaPipeline.run(GeMoMaPipeline.java:1038)
	at projects.gemoma.GeMoMaModule.run(GeMoMaModule.java:94)
	at de.jstacs.tools.ui.cli.CLI.run(CLI.java:426)
	at projects.gemoma.GeMoMa.main(GeMoMa.java:399)


I assume it could be that the synteny check is making the problems as Augustus predictions cannot be used for synteny check (#43, #21). Please try to run with sc=false

However, I also noticed that the result are somehow wired. Although you have enough transcripts extracted from the reference species (12-24K transcripts per reference species), the GAF module filters out most of them:

all: 605809
filtered: 16144
clustered: 9027

...

input	filtered transcripts	final transcripts supported	final genes supported
input 0	106	89	89
input 1	203	158	157
input 2	248	72	72
input 3	166	117	117
input 4 (august)	15594	8839	8839


This seems to be wired. Less than 250 transcripts per reference species passed the filter. It seems that you haven't modified the filter, i.e. it is the default value. With default value (start=='M' and stop=='*' and (isNaN(score) or score/aa>=0.75)) it filters for having a start and a stop codon and for some conservation (score/aa>=0.75). I would recommend to check why so many predictions were discarded.

I would recommend to run GeMoMaPipeline again with additional parameter setting sc=false. You already used o=true. This allows to run GAF with different parameter settings after the GeMoMaPipeline.
